cura/OAuth2/AuthorizationService.py

```diff
@@ -87,19 +87,20 @@ def _parseJWT(self) -> Optional["UserProfile"]:
 
         return self._auth_helpers.parseJWT(self._auth_data.access_token)
 
-    #   Get the access token as provided by the repsonse data.
+    #   Get the access token as provided by the response data.
     def getAccessToken(self) -> Optional[str]:
-        if not self.getUserProfile():
-            # We check if we can get the user profile.
-            # If we can't get it, that means the access token (JWT) was invalid or expired.
-            Logger.log("w", "Unable to get the user profile.")
-            return None
-
         if self._auth_data is None:
             Logger.log("d", "No auth data to retrieve the access_token from")
             return None
 
-        return self._auth_data.access_token
+        # Check if the current access token is expired and refresh it if that is the case.
+        received_at = datetime.strptime(self._auth_data.received_at, TOKEN_TIMESTAMP_FORMAT) \
+            if self._auth_data.received_at else datetime(2000, 1, 1)
```

Comment:
Shouldn't we store the expiration date instead? And handle it nicely if the token expires..

Reply:
We must handle 401's correctly anyway, as the token may be unvalidated by the server for other reasons